packages/aws-cdk-lib/aws-dynamodb/lib/table-v2.ts

```diff
@@ -664,8 +664,9 @@ export class TableV2 extends TableBaseV2 {
   private configureReplicaTable(props: ReplicaTableProps): CfnGlobalTable.ReplicaSpecificationProperty {
     const pointInTimeRecovery = props.pointInTimeRecovery ?? this.tableOptions.pointInTimeRecovery;
     const contributorInsights = props.contributorInsights ?? this.tableOptions.contributorInsights;
-    const resourcePolicy = props.resourcePolicy ?? this.tableOptions.resourcePolicy;
-
+    //const resourcePolicy = props.resourcePolicy ?? this.tableOptions.resourcePolicy;
+    //const resourcePolicy = props.resourcePolicy;
+    const resourcePolicy = (props.region === this.region ? this.tableOptions.resourcePolicy : props.resourcePolicy) || undefined;
```

Comment:
This is a breaking change for existing users. If existing users manage to bypass the CFN error by using escape hatch. They then remove the escape hatch and upgrade to a version that includes this change, their replication region will now have the resource policies removed, thus this is not an acceptable change. I would recommend you to do it following the suggested proposal in the original issue.
```
Ideally, the construct could allow null and when null is specified in a specific replica, no resourcePolicy is added to that replica even when one is defined in the TableV2 itself

new TableV2(this, `MyTable-${stage}`, {
      pointInTimeRecovery: true,
      partitionKey: {
        name: 'key',
        type: AttributeType.STRING,
      },
      tableName: 'MyTable',
      // us-east-2 should not have tablePolicyDocument added in the template
      replicas: [{ region: 'us-east-1' }, { region: 'us-east-2', resourcePolicy: null }], 
      resourcePolicy: tablePolicyDocument,
    });

```

Reply:
This is forcing a user to add null for when they don't want a resource policy. The default implementation should only require them to add a policy to a replica should the want it. It shouldn't default to adding it to all replicas.